Cloudrig hierarchy fix

AbsoluteMoose opened the pull request:
Rework the logic of the Rigify Hierarchy Fix to better handle rigs created by the Blender Studio [CloudRig](https://extensions.blender.org/add-ons/cloudrig/) addon.
- Removes redundant code
- Fixes situations where bones are indirectly parented due to mechanism bones
- Fixes situations where bones parented to limbs may be parented incorrectly due to their parent being split into multiple twist bones with a _# suffix

Requires #4

AbsoluteMoose committed:
Add options to allow extracting bones within a bone collection

Adds an extract mode option to the deform rig generator that allows the user to specify a bone collection containing bones to be extracted for the game rig. "Collection" mode extracts only bones within the collection, while "Deform and Collection" mode extracts deform bones as well as bones within the collection. This is useful for generating rigs that require the inclusion of non-deform bones for the purposes of in-game IK or procedural animation.

BlenderBoi commented:
thanks for the contribution, I need some time to review the code, so I hope you don't mind

Also, I kinda have plans to rewrite some part of the addon to work better in blender 5.0 

so just letting you know, in case your hardwork get replaced in future

AbsoluteMoose commented:
These changes were primarily for personal ease of use, but I figured they were generally useful enough to warrant a pull request. Neither are particularly complex logic-wise, so if 5.0 changes the overall code significantly they shouldn't be too difficult to re-implement.

BlenderBoi commented:
No problem, just letting you know just in case

Appreciate the contribution tho

AbsoluteMoose committed:
Cloudrig Hierarchy Fix

Rework the logic of the Rigify Hierarchy Fix to better handle rigs created by the Blender Studio CloudRig addon.

- Removes redundant code
    Fixes situations where bones are indirectly parented due to mechanism bones
- Fixes situations where bones parented to limbs may be parented incorrectly due to their parent being split into multiple twist bones with a _# suffix

AbsoluteMoose committed:
Improve stretch bone handling

- Fixes cases where deform bones are parented to stretch bones using armature constraints
- Moves get_included_parent() declaration to within execute() in order to avoid needing bone lists as parameters
